### DISCUSSION TITLE
Initializing Kraken 360 (2023 standard)

### DISCUSSION THREAD

**Discussion by rossg5309** [liquidctl/liquidctl#652] [General]
I'm having issues getting liquidctl to recognize my liquid cooler. I was able to locate the device with `lsusb -vt `, and attempted a previously recommended fix by manually rebinding the device with `echo '1-6.1:1.1' | sudo tee /sys/bus/usb/drivers/usbhid/bind`, but I am returned `Device or resource busy`, which leaves me still dead in the water. Any recommendations welcome.

I currently have the V2 smart device and Kraken routed through [NZXTs Internal USB hub ](https://nzxt.com/product/internal-usb-hub-gen-3). Everything works fine in Windows with NZXT Cam.

Contents of `liquidctl list --verbose`:

Device #0: NZXT Smart Device V2
├── Vendor ID: 0x1e71
├── Product ID: 0x2006
├── Release number: 0x0200
├── Serial number: 00000000001A
├── Bus: hid
├── Address: /dev/hidraw5
└── Driver: SmartDevice2

Device #1: ASUS Aura LED Controller
├── Vendor ID: 0x0b05
├── Product ID: 0x18f3
├── Release number: 0x0100
├── Serial number: 9876543210
├── Bus: hid
├── Address: /dev/hidraw0
└── Driver: AuraLed

Contents of `liquidctl list --debug`:

[DEBUG] (cli) (_log_env_infos): script: /home/ross/miniconda3/bin/liquidctl
[DEBUG] (cli) (_log_env_infos): version: 1.13.0
[DEBUG] (cli) (_log_env_infos): platform: Linux-6.2.0-37-generic-x86_64-with-glibc2.35
[DEBUG] (cli) (_log_env_infos): python: 3.11.5 (main, Sep 11 2023, 13:54:46) [GCC 11.2.0]
[DEBUG] (cli) (_log_env_infos): encoding: UTF-8 current, UTF-8 preferred, utf8_mode 0
[DEBUG] (cli) (_log_env_infos): with colorlog: 6.7.0
[DEBUG] (cli) (_log_env_infos): with crcmod: 1.7
[DEBUG] (cli) (_log_env_infos): with docopt: 0.6.2
[DEBUG] (cli) (_log_env_infos): with hidapi: 0.14.0
[DEBUG] (cli) (_log_env_infos): with pyusb: 1.2.1
[DEBUG] (cli) (_log_env_infos): with pillow: 10.1.0
[DEBUG] (cli) (_log_env_infos): with smbus: 1.1.post2
[DEBUG] (cli) (_log_env_infos): with winusbcdc: version n/a (No package metadata was found for winusbcdc)
[DEBUG] (cli) (_log_env_infos): with libusb-package: version n/a (No package metadata was found for libusb-package)
[DEBUG] (smbus) (find_devices): searching LinuxI2c
[DEBUG] (smbus) (find_devices): LinuxI2c drivers: Ddr4Temperature, EvgaPascal, RogTuring, VengeanceRgb
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-3
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-10
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-1
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-8
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-6
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-4
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-11
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-2
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-0
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-9
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-7
[DEBUG] (smbus) (find_devices): I²C adapter: i2c-5
[DEBUG] (usb) (find_devices): searching HidapiBus
[DEBUG] (usb) (find_devices): HidapiBus drivers: Aquacomputer, AuraLed, CommanderCore, CommanderPro, CorsairHidPsu, H1V2, HydroPlatinum, Kraken2, KrakenX3, KrakenZ3, NzxtEPsu, RgbFusion2, SmartDevice, SmartDevice2, _BaseSmartDevice
[DEBUG] (usb) (find_devices): HID device: 1e71:300e (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 1e71:2006 (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (__init__): has kernel driver: nzxt-smart2 (/sys/class/hidraw/hidraw5/device/hwmon/hwmon4)
[DEBUG] (usb) (probe): SmartDevice2 identified: NZXT Smart Device V2
[DEBUG] (usb) (find_devices): HID device: 046d:c332 (usage_page=0x0001 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 046d:c332 (usage_page=0x0001 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:c332 (usage_page=0x0001 usage=0x0006)
[DEBUG] (usb) (find_devices): HID device: 046d:c332 (usage_page=0x000c usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:c332 (usage_page=0x0001 usage=0x0080)
[DEBUG] (usb) (find_devices): HID device: 046d:c332 (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 046d:c332 (usage_page=0xff00 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 0c45:7016 (usage_page=0x0001 usage=0x0006)
[DEBUG] (usb) (find_devices): HID device: 0c45:7016 (usage_page=0x0001 usage=0x0002)
[DEBUG] (usb) (find_devices): HID device: 0c45:7016 (usage_page=0x0001 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 0c45:7016 (usage_page=0x0001 usage=0x0080)
[DEBUG] (usb) (find_devices): HID device: 0c45:7016 (usage_page=0x000c usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 0c45:7016 (usage_page=0x0001 usage=0x0006)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0x000c usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x0001)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x000a)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x0022)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x000a)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x0022)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x000a)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x0022)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x000a)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x0022)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x000a)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x000c)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0x0001 usage=0x003a)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0xff00 usage=0x000c)
[DEBUG] (usb) (find_devices): HID device: 1038:1297 (usage_page=0x0001 usage=0x003a)
[DEBUG] (usb) (find_devices): HID device: 0b05:18f3 (usage_page=0xff72 usage=0x00a1)
[DEBUG] (usb) (probe): AuraLed identified: ASUS Aura LED Controller
[DEBUG] (usb) (find_devices): searching PyUsbBus
[DEBUG] (usb) (find_devices): PyUsbBus drivers: Hydro690Lc, HydroPro, Legacy690Lc, Modern690Lc, _Base690Lc, _ModernBase690Lc
[DEBUG] (usb) (find_devices): USB device: 1d6b:0003
[DEBUG] (usb) (find_devices): USB device: 1d6b:0002
[DEBUG] (usb) (find_devices): USB device: 0bda:0411
[DEBUG] (usb) (find_devices): USB device: 0bda:0411
[DEBUG] (usb) (find_devices): USB device: 1d6b:0003
[DEBUG] (usb) (find_devices): USB device: 0b05:18f3
[DEBUG] (usb) (find_devices): USB device: 0bda:5411
[DEBUG] (usb) (find_devices): USB device: 1038:1297
[DEBUG] (usb) (find_devices): USB device: 0451:2036
[DEBUG] (usb) (find_devices): USB device: 0c45:7016
[DEBUG] (usb) (find_devices): USB device: 046d:c332
[DEBUG] (usb) (find_devices): USB device: 0bda:5411
[DEBUG] (usb) (find_devices): USB device: 1d6b:0002
[DEBUG] (usb) (find_devices): USB device: 1d6b:0003
[DEBUG] (usb) (find_devices): USB device: 1e71:2006
[DEBUG] (usb) (find_devices): USB device: 1e71:300e
[DEBUG] (usb) (find_devices): USB device: 05e3:0608
[DEBUG] (usb) (find_devices): USB device: 05e3:0608
[DEBUG] (usb) (find_devices): USB device: 1d6b:0002
Device #0: NZXT Smart Device V2
├── Vendor ID: 0x1e71
├── Product ID: 0x2006
├── Release number: 0x0200
├── Serial number: 00000000001A
├── Bus: hid
├── Address: /dev/hidraw5
└── Driver: SmartDevice2
[DEBUG] (cli) (_list_devices_human): MRO: SmartDevice2, _BaseSmartDevice, UsbHidDriver, BaseUsbDriver, BaseDriver, object

Device #1: ASUS Aura LED Controller
├── Vendor ID: 0x0b05
├── Product ID: 0x18f3
├── Release number: 0x0100
├── Serial number: 9876543210
├── Bus: hid
├── Address: /dev/hidraw0
└── Driver: AuraLed
[DEBUG] (cli) (_list_devices_human): MRO: AuraLed, UsbHidDriver, BaseUsbDriver, BaseDriver, object

Truncated contents of `lsusb -vt`:
/:  Bus 01.Port 1: Dev 1, Class=root_hub, Driver=xhci_hcd/6p, 480M
    ID 1d6b:0002 Linux Foundation 2.0 root hub
    |__ Port 5: Dev 2, If 0, Class=Hub, Driver=hub/4p, 480M
        ID 05e3:0608 Genesys Logic, Inc. Hub
    |__ Port 6: Dev 3, If 0, Class=Hub, Driver=hub/4p, 480M
        ID 05e3:0608 Genesys Logic, Inc. Hub
        |__ Port 3: Dev 5, If 0, Class=Human Interface Device, Driver=usbhid, 12M
            ID 1e71:2006 NZXT Smart Device V2
        |__ Port 1: Dev 4, If 1, Class=Human Interface Device, Driver=usbhid, 480M
            ID 1e71:300e NZXT 
        |__ Port 1: Dev 4, If 0, Class=Vendor Specific Class, Driver=, 480M
            ID 1e71:300e NZXT

**Comment by aleksamagicka** [comment]
Support for that Kraken in liquidctl has been merged in [october](https://github.com/liquidctl/liquidctl/commit/846ac5f5aa9d5484e8ed1d549fd2478bbf51fdd8) and [v1.13.0](https://github.com/liquidctl/liquidctl/releases/tag/v1.13.0) is from July. Generally, please refer to the [compatibility table](https://github.com/liquidctl/liquidctl#supported-devices).

Please retry with the [Git version](https://github.com/liquidctl/liquidctl#installing-from-pypi-or-github) for the time being.

**Reply by rossg5309** [reply]
Well that explains a hell of a lot. Got it working. Cheers.
